File: GC.WebSpace/ClientApp/domain/gardens/sectorSales/sectorSaleBlank.ts
import { SectorSale } from "./sectorSale";

export class SectorSaleBlank {
    constructor(
        public id: string | null,
        public sectorId: string,
        public firstName: string,
        public lastName: string,
        public middleName: string | null,
        public description: string | null,
        public price: number | null,
        public phoneNumber: string
    ) { }

    public static empty = () => {
        return new SectorSaleBlank(null, "", "", "", null, null, null, "");
    }

    public static create = (sale: SectorSale) => {
        return new SectorSaleBlank(sale.id, sale.sector.id, sale.firstName, sale.lastName, sale.middleName, sale.description,
            sale.price, sale.phoneNumber);
    }
}

File: GC.Domain/Gardens/Sectors/Sales/SectorSaleBlank.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GC.Domain.Gardens.Sectors.Sales
{
    public class SectorSaleBlank
    {
        public Guid? Id { get; set; }
        public Guid? SectorId { get; set; }
        public String FirstName { get; set; }
        public String LastName { get; set; }
        public String MiddleName { get; set; }
        public String Description { get; set; }
        public Int32? Price { get; set; }
        public String PhoneNumber { get; set; }
    }
}
